### PR TITLE
ref(ui) Convert newsletter form to new form components

### PR DESCRIPTION
**Pull request by markstory**
Minor layout changes from using the new components. I'm not sure about the alignment of the paragraph and form fields. I think we might need a field component that contains text to make these kinds of layouts work better.

### Before

Screen Shot 2021-09-22 at 8 52 33 PM

### After

Screen Shot 2021-09-22 at 8 49 40 PM

### CODE REVIEW

**Review by priscilawebdev — COMMENTED** [review]
yes I agree... I think we should fix this layout because now it looks a little odd. Are we planning to revisit this?

File: static/app/views/newsletterConsent.tsx [hunk 1]
```diff
@@ -22,35 +23,37 @@ export default class NewsletterConsent extends Component<Props> {
...
         <ApiForm
           apiMethod="POST"
           apiEndpoint="/users/me/subscriptions/"
           onSubmitSuccess={() => this.props.onSubmitSuccess?.()}
           submitLabel={t('Continue')}
         >
-          <RadioBooleanField
-            key="subscribed"
+          <p>
+            {t('Pardon the interruption, we just need to get a quick answer from you.')}
```
Comment (priscilawebdev) [inline static/app/views/newsletterConsent.tsx:33]: nit: should we replace `Pardon` with `Sorry`? :D

Comment (markstory) [reply static/app/views/newsletterConsent.tsx:33]: We could, but the phrasing would need to change to something like `Sorry to interrupt, we just...`.

**Comment by markstory** [comment]
I've used `FieldWrapper` to improve the text alignment and it now looks like.

Screen Shot 2021-09-23 at 10 05 21 AM

**Review by priscilawebdev — APPROVED** [review]
it looks better now...thanks

File: static/app/views/newsletterConsent.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/newsletterConsent.tsx:34]: wouldn't it be better? :)

**Pull request merged**